preview-src/context.tsx

```diff
@@ -0,0 +1,171 @@
+		}
+	}
+
+	public setTitle = (title: string) =>
+		this.postMessage({ command: 'pr.edit-title', args: { text: title } })
+
+	public checkout = () =>
+		this.postMessage({ command: 'pr.checkout' })
+
+	public exitReviewMode = async () => {
+		if (!this.pr) { return; }
+		return this.postMessage({
+			command: 'pr.checkout-default-branch',
+			args: this.pr.repositoryDefaultBranch,
+		});
+	}
+
+	public refresh = () =>
+		this.postMessage({ command: 'pr.refresh' })
+
+	public merge = (args: { title: string, description: string, method: MergeMethod }) =>
+		this.postMessage({ command: 'pr.merge', args	})
+
+	public comment = async (args: string) => {
+		const result = await this.postMessage({ command: 'pr.comment', args});
+		const newComment = result.value;
+		newComment.event = EventType.Commented;
+		this.updatePR({
+			events: [...this.pr.events, newComment],
+			pendingCommentText: '',
+		});
+	}
+
+	public addReviewers = () =>
+		this.postMessage({ command: 'pr.add-reviewers' })
+
+	public addLabels = () =>
+		this.postMessage({ command: 'pr.add-labels' })
+
+	public deleteComment = async (args: { id: number, pullRequestReviewId?: number }) => {
+		await this.postMessage({ command: 'pr.delete-comment', args });
+		const { pr } = this;
+		const { id, pullRequestReviewId } = args;
+		if (!pullRequestReviewId) {
+			this.updatePR({
+				events: pr.events.filter(e => e.id !== id)
+			});
+			return;
+		}
+		const index = pr.events.findIndex(e => e.id === pullRequestReviewId);
+		if (index === -1) {
+			console.error('Could not find review:', pullRequestReviewId);
+			return;
+		}
+		const review: ReviewEvent = pr.events[index] as ReviewEvent;
+		if (!review.comments) {
+			console.error('No comments to delete for review:', pullRequestReviewId, review);
+			return;
+		}
+		this.pr.events.splice(index, 1, {
+			...review,
+			comments: review.comments.filter(c => c.id !== id)
+		});
+		this.updatePR(this.pr);
+	}
+
+	public editComment = (args: {comment: Comment, text: string}) =>
+		this.postMessage({ command: 'pr.edit-comment', args })
+
+	public updateDraft = (id: number, body: string) => {
+		let pullRequest = getState();
+		const pendingCommentDrafts = pullRequest.pendingCommentDrafts || Object.create(null);
+		if (body === pendingCommentDrafts[id]) { return; }
+		pendingCommentDrafts[id] = body;
+		this.updatePR({ pendingCommentDrafts: pendingCommentDrafts });
+	}
+
+	public requestChanges = async (body: string) =>
+		this.appendReview(await this.postMessage({ command: 'pr.request-changes', args: body }))
+
+	public approve = async (body: string) =>
+		this.appendReview(await this.postMessage({ command: 'pr.approve', args: body }))
+
+	public submit = async (body: string) =>
+		this.appendReview(await this.postMessage({ command: 'pr.submit', args: body }))
+
+	public removeReviewer = async (review: ReviewState) => {
+		await this.postMessage({ command: 'pr.remove-reviewer', args: review });
```

Comment:
We only need to send the login, not the whole reviewer object. I get this error `Invalid request.

For 'items', {"reviewer"=>{"avatarUrl"=>"https://avatars3.githubusercontent.com/u/3672607?v=4", "login"=>"RMacfarlane", "url"=>"https://github.com/RMacfarlane"}, "state"=>"REQUESTED"} is not a string.`

Reply:
Fixed.